src/components/question/question.tsx

```diff
@@ -0,0 +1,105 @@
+import './question.css'
+
+import { useMediaQuery, useTheme } from '@mui/material'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import React, { useState } from 'react'
+import { v4 as getUUID } from 'uuid'
+
+import Icon from '../icon'
+import MarkedMarkdown from '../markdown/markedMarkdown'
+import type { Message, QuestionData } from '../types'
+
+/**
+'The `Question` component provides a user interface for selecting an answer from a list of choices. It is designed to facilitate interactive decision-making and response submission within a conversation or messaging context.
+ */
+export default function Question(props: QuestionData) {
+  const [selectedAnswer, setSelectedAnswer] = useState('')
+
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
+  const buttonGroupOrientation = isMobile ? 'column' : 'row'
+
+  function handleSubmitResponse(response: string) {
+    setSelectedAnswer(response)
+    const currentTime = new Date().toISOString()
+
+    const formattedMessage: Message = {
+      id: getUUID(),
+      timestamp: currentTime,
+      sender: props.sender,
+      conversationId: props.conversationId,
+      format: 'text',
+      data: { text: response },
+      inReplyTo: props.messageId,
+    }
+
+    props.ws.send(formattedMessage)
+  }
+
+  const buttonList = props.answers.map((answer) => {
+    let buttonStyles = {
+      borderColor: 'secondary.dark',
+      color: 'text.primary',
+    }
+
+    const selectedStyles = {
+      backgroundColor: 'secondary.main',
+      '&.Mui-disabled': { color: 'text.primary' },
+    }
+
+    if (answer.label === selectedAnswer) {
+      buttonStyles = {
+        ...buttonStyles,
+        ...selectedStyles,
+      }
+    }
+
+    return (
+      <Button
+        key={answer.label}
+        onClick={() => handleSubmitResponse(answer.label)}
+        variant="outlined"
+        sx={buttonStyles}
+        startIcon={selectedAnswer === answer.label && <Icon name="check" />}
+        className="rustic-answer-button"
+        disabled={!!selectedAnswer}
+      >
+        {answer.label}
+      </Button>
+    )
+  })
+
+  return (
+    <Box className="rustic-question" data-cy="question">
+      {(props.title || props.description) && (
+        <Box className="rustic-question-text">
+          {props.title && (
+            <Typography variant="subtitle2" className="rustic-title">
+              {props.title}
+            </Typography>
+          )}
+
+          {props.description && <MarkedMarkdown text={props.description} />}
+        </Box>
+      )}
+
+      {props.answers.length > 0 ? (
+        <Stack
+          direction={buttonGroupOrientation}
+          data-cy="buttons-container"
```

Comment:
I'll add this to avoid overflow. But I'm wondering how we'll want buttons to appear in cases like this. @Laurendragonscale @RenataDzotova 

<img width="475" alt="Screenshot 2024-05-24 at 3 31 32 PM" src="https://github.com/rustic-ai/ui-components/assets/111031789/572ba393-6af8-4882-891a-d36943ac6305">